src/pages/home/Home.tsx

```diff
@@ -7,23 +7,37 @@ import { useGetUserStudentsQuery } from "../../app/services/UserService";
 import useAuthentication from "../../hooks/useAuthentication";
 import { useDeleteStudentMutation } from "../../app/services/StudentService";
 import toast from "react-hot-toast";
+import { ConfirmDialog } from "../../components/confirmDialog/ConfirmDialog"; 
+import { useState } from "react";
 
 function Home() {
   const { userId } = useAuthentication();
   const { data: students } = useGetUserStudentsQuery(userId!);
   const [deleteStudent] = useDeleteStudentMutation();
   const navigate = useNavigate();
+  const [confirmDialogOpen, setConfirmDialogOpen] = useState(false);
+  const [studentToDelete, setStudentToDelete] = useState<{id: string, name: string} | null>(null);
 
   const handleDelete = async (studentId: string, studentName: string) => {
-    const confirmDelete = window.confirm(`¿Desea eliminar a ${studentName}?`);
-    if (!confirmDelete) return;
-    deleteStudent(studentId)
+    setStudentToDelete({ id: studentId, name: studentName });
+    setConfirmDialogOpen(true);
+  };
+  const confirmDelete = async () => {
```

Comment:
Los nombres de los metodos cuando se reacciona a un evento, suelen seguir la siguiente convencion


```suggestion
  const handleConfirmDeleteClick = async () => {
```